discovery-provider/src/tasks/index_rewards_manager.py

```diff
@@ -257,15 +294,26 @@ def process_batch_sol_reward_manager_txs(
             .all()
         )
         user_challenge_specifiers = {challenge[0] for challenge in user_challenges}
+        challenges = (
+            session.query(Challenge.id, Challenge.type)
+            .filter(bool(Challenge.active))
+            .all()
+        )
+        challenge_type_map = {
+            challenge[0]: TransactionType.trending_reward
+            if challenge[1] == ChallengeType.trending
+            else TransactionType.user_reward
+            for challenge in challenges
+        }
```

Comment:
This map should be the same for each pass through indexing, right? We should be able to avoid doing another DB roundtrip for this – maybe pull it into a function that calls it once if it doesn't exist and otherwise returns the stored value or smth